test/js/bun/shell/throw.test.ts

```diff
@@ -0,0 +1,50 @@
+import { $ } from "bun";
+import { beforeAll, describe, test, expect } from "bun:test";
+
+beforeAll(() => {
+  $.nothrow();
+});
+
+describe("throw", () => {
+  test("enabled globally", async () => {
+    $.throws(true);
+    let e;
+    try {
+      await $`ls ksjflkjfksjdflksdjflksdf`;
+      expect("Woops").toBe("Should have thrown");
+    } catch (err) {
+      e = err;
+    }
+    expect(e).toBeDefined();
+  });
+
+  test("enabled locally", async () => {
+    let e;
+    try {
+      await $`ls ksjflkjfksjdflksdjflksdf`.throws(true);
+      expect("Woops").toBe("Should have thrown");
+    } catch (err) {
+      e = err;
+    }
+    expect(e).toBeDefined();
+  });
+
+  test("disable globally", async () => {
+    $.throws(true);
+    $.nothrow();
```

Comment:
So first we turn throwing on and then off?

Reply:
Not necessary, I'm just doing it here to be extra careful with testing behaviour since `$.nothrow()` is in the `beforeAll`